x/ecocredit/spec/README.md

```diff
@@ -1,10 +1,46 @@
 # Ecocredit Module
 
-## Abstract
+::: tip Now Available
+The first version of the ecocredit module was made available in Regen Ledger v2.0.
+:::
 
-The following documents specify the ecocredit module.
+The following documentation provides a technical overview of the ecocredit module and is designed for developers building tools and services that interact with Regen Ledger.
 
-The ecocredit module enables the ability to manage classes of ecosystem service credits (i.e. carbon credits, biodiversity credits, soil health credits, etc), defined as "credit classes" and to mint credits through a batch issuance process.
+For more information about specific ecosystem service credits and methodologies being issued and developed through the Regen Registry Program, please see [Regen Registry Library](https://library.regen.network/).
+
+## Overview
+
+The ecocredit module is divided into three separate components, including "core functionality" and two submodules, the "basket submodule" and the "marketplace submodule".
+
+### Core Functionality
+
+::: tip Now Available
+The core functionality was made available in Regen Ledger v2.0. Added support for projects will be made available in Regen Ledger v4.0.
+:::
+
+The core functionality of the ecocredit module enables the creation and management of credit types, credit classes, and projects, as well as the issuance, transferring, and retirement of ecosystem service credits (i.e. carbon credits, biodiversity credits, soil health credits, etc.).
+
+Credit classes are designed to support a variety of ecosystem service credits, and can be created and managed by individuals, groups, or organizations. Each credit class is associated with a credit type that represents the primary unit of measurement for the methodologies defined within the credit class (i.e. carbon, biodiversity, etc.). Credit types can only be added through on-chain governance.
+
+Ecosystem service credits are issued through a batch issuance process where only approved issuers for a given credit class can issue credits from that credit class. Each credit batch is associated with a project representing the on-the-ground project implementing the methodologies defined within the credit class. Credit batches are unique in that each credit batch has a start and end date representing the period in which the positive ecological outcomes were measured.
+
+Credits issued within a credit batch are only fungible with credits from the same batch. Credits can be issued in a retired or tradable state. Retiring a credit implies the credit is being consumed as an offset. Credits issued in a tradable state can be transferred or retired by the owner.
```

Comment:
i added a warning in the godoc of `Retire` that it's permanent. should we add that here?

Reply:
Good idea. I can add.

Reply:
https://github.com/regen-network/regen-ledger/pull/914/commits/42c66578370c03d2eb067a3775dc2457317f1e13